### ISSUE TITLE
[BUG] Unable to pull image from ACR when using a managed identity for AKS

### ISSUE THREAD

**Issue by googol** [SparebankenVest/azure-key-vault-to-kubernetes#85]
**Describe the bug**
I've been playing around with akv2k8s and AKS managed identity today, and came up with an error. The webhook is not able to pull the image for my container from ACR, and thus doesn't mutate. Then my container starts up without the proper env variables and enters into a crash loop.

I've given the kubelet identity the AcrPull role, and the nodes are able to pull the images from ACR and start the pods. 

**To Reproduce**
Do the getting started steps on an AKS cluster with managed identity enabled (https://docs.microsoft.com/en-us/azure/aks/use-managed-identity)

**Expected behavior**
The webhook is able to mutate the container to use the env injector

**Logs**
```
 time="2020-06-05T11:40:58Z" level=info msg="found pod to mutate in namespace 'default'"                                                                                                                                                                    
 time="2020-06-05T11:40:58Z" level=info msg="found container 'CONTAINER' to mutate"                                                                                                                                                                          
 time="2020-06-05T11:40:58Z" level=info msg="checking for env vars containing '@azurekeyvault' in container CONTAINER"                                                                                                                                       
 time="2020-06-05T11:40:58Z" level=info msg="found env var: SECRET@azurekeyvault"                                                                                                                                                       
 time="2020-06-05T11:40:58Z" level=info msg="did not find credentials to use with registry 'ACR.azurecr.io' - getting default credentials"                                                                                                             
 time="2020-06-05T11:40:58Z" level=info msg="using default credentials for docker registry with clientid: msi"                                                                                                                                              
 time="2020-06-05T11:40:58Z" level=info msg="pulling docker image ACR.azurecr.io/CONTAINER:latest to get entrypoint and cmd, timeout is 120 seconds"                                                                                                    
 2020/06/05 11:40:59 [ERROR] admission webhook error: failed to get auto cmd, error: failed to pull docker image 'ACR.azurecr.io/CONTAINER:latest', error: Error response from daemon: Get https://ACR.azurecr.io/v2/CONTAINER/manifests/latest: unauthorized: Application not registered with AAD.                                                                                                                                                                                                           
```

**Additional context**

**Comment by googol** [comment]
Hey @torresdal, sorry to ping you, I just noticed you had responded to some other recent issues and wanted to check if this is something you guys could take a look at

**Comment by googol** [comment]
Seems to be the same as #89

**Comment by JakubSurfer** [comment]
I faced the same problem so I would really appreciate some input here.

**Comment by torresdal** [comment]
@googol This will be available in the upcoming 1.1 version - ref #110
